File: cpp/idades.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    string nome, nome2;
    int idade, idade2;
    double media;

    cout << "Dados da primeira pessoa: " << endl;

    cout << "Nome: ";
    getline(cin, nome);

    cout << "Idade: ";
    cin >> idade;

    cout << "Dados da segunda pessoa: " << endl;

    cout << "Nome: ";
    cin.ignore(INT_MAX, '\n');
    getline(cin, nome2);

    cout << "Idade: ";
    cin >> idade2;

    media = (double) (idade + idade2) / 2;

    cout << "A idade media de " << nome << " e " << nome2 << " eh de " << media << " anos.";
    return 0;
}

File: java/idades.java
import java.util.Locale;
import java.util.Scanner;

public class main {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		System.out.println("Dados da primeira pessoa:");
		System.out.print("Nome: ");
		String nome1 = sc.nextLine();
		System.out.print("Idade: ");
		int idade1 = sc.nextInt();
		
		
		System.out.println("Dados da segunda pessoa:");
		sc.nextLine();
		System.out.print("Nome: ");
		String nome2 = sc.nextLine();
		System.out.print("Idade: ");
		int idade2 = sc.nextInt();
		
		double media = (double) (idade1 + idade2) / 2;
		
		System.out.println("A idade media de " + nome1 + " e " + nome2 + " eh de " + media + " anos.");
		
		
		
		
		sc.close();

	}

}
